ui/apps/platform/src/Containers/DeferralConfiguration/VulnerabilitiesConfiguration.tsx

```diff
@@ -9,54 +10,148 @@ import {
-                    <Switch label="Enabled" labelOff="Disabled" isChecked={isEnabled} />
+                    <Switch
+                        id={`${fieldId}.enabled`}
+                        label="Enabled"
+                        labelOff="Disabled"
+                        isChecked={isEnabled}
+                        onChange={(_, e) => handleChange(e)}
+                    />
                 </FormGroup>
             </GridItem>
         </>
     );
 }
 
-function BooleanSetting({ label, isEnabled }: BaseSettingProps & { label: string }) {
+function BooleanSetting({
+    fieldId,
+    label,
+    isEnabled,
+    handleChange,
+}: BaseSettingProps & {
+    label: string;
+}) {
     return (
         <>
             <GridItem className="pf-u-py-xs" span={8} md={4} xl={3}>
                 <p>{label}</p>
             </GridItem>
             <GridItem className="pf-u-py-xs" span={4} md={8} xl={9}>
                 <FormGroup>
-                    <Switch label="Enabled" labelOff="Disabled" isChecked={isEnabled} />
+                    <Switch
+                        id={fieldId}
+                        label="Enabled"
+                        labelOff="Disabled"
+                        isChecked={isEnabled}
+                        onChange={(_, e) => handleChange(e)}
+                    />
                 </FormGroup>
             </GridItem>
         </>
     );
 }
+function getDefaultConfig(): VulnerabilitiesDeferralConfig {
+    return {
+        expiryOptions: {
+            dayOptions: [
+                { numDays: 1, enabled: false },
+                { numDays: 1, enabled: false },
+                { numDays: 1, enabled: false },
+                { numDays: 1, enabled: false },
+            ],
+            fixableCveOptions: {
+                allFixable: false,
+                anyFixable: false,
+            },
+            customDate: false,
+        },
+    };
+}
+
+function ensureMinimumDayOptions(
+    config: VulnerabilitiesDeferralConfig
+): VulnerabilitiesDeferralConfig {
+    const minimumLength = 4;
+    const dayOptions = [...config.expiryOptions.dayOptions];
+    while (dayOptions.length < minimumLength) {
+        dayOptions.push({ numDays: 1, enabled: false });
+    }
+
+    return {
+        ...config,
+        expiryOptions: {
+            ...config.expiryOptions,
+            dayOptions: sortBy(dayOptions, 'numDays'),
+        },
+    };
+}
 
 function VulnerabilitiesConfiguration() {
+    const { config, isConfigLoading, isUpdateInProgress, configLoadError } =
+        useVulnerabilitiesDeferralConfig();
+
+    const { values, handleChange } = useFormik({
+        enableReinitialize: true,
+        // Ensure that there are at least 4 day options in case this array was set to zero via the API
+        initialValues: ensureMinimumDayOptions(config ?? getDefaultConfig()),
+        onSubmit: () => {},
+    });
+
+    const isConfigDirty = !isEqual(ensureMinimumDayOptions(config ?? getDefaultConfig()), values);
```

Comment:
We can probably extract out the `ensureMinimumDayOptions(config ?? getDefaultConfig())` into a variable to be used here and line `147`

Reply:
Oh good call, I had these separate before I added the `enableReinitialization` prop to Formik, so these might be able to be collapse into a single variable now. I'll test it out and update if so.